test/unit/modules/work.test.ts

```diff
@@ -0,0 +1,32 @@
+import { Work } from '../../../lib/modules/work/work';
+
+describe('Work class', () => {
+  let work: Work;
+  let rpcMock: any;
+
+  beforeEach(() => {
+    rpcMock = {
+      call: jest.fn(),
+    };
+    work = new Work(rpcMock);
+  });
+
+  describe('generate', () => {
+    it('should call the RPC method with the correct parameters and return the generated work', async () => {
+      const hash = 'abc123';
+      const generatedWork = {
+        work: 'work',
+        difficulty: 'difficulty',
```

Comment:
Maybe use realistic values (e.g. "fffffff93c41ec94" for difficulty and "1.182623871097636" based on the [Nano docs](https://docs.nano.org/commands/rpc-protocol/#work_generate)